src/common/base-store.ts

```diff
@@ -33,7 +33,7 @@ import { kebabCase } from "lodash";
 import { getPath } from "./utils/getPath";
 
 export interface BaseStoreParams<T> extends ConfOptions<T> {
-  syncOptions?: IReactionOptions;
+  syncOptions?: IReactionOptions<T, boolean>;
```

Comment:
Maybe we should simplify / define own interface for `syncOptions`?

New `mobx@6.3.5 ` definition for `IReactionOptions` looks weird:

```ts
export type IReactionOptions<T, FireImmediately extends boolean> = IAutorunOptions & {
    fireImmediately?: FireImmediately
    equals?: IEqualsComparer<T>
}
```

We could keep it something as in format before:

```ts
import { IAutorunOptions, IEqualsComparer } from "mobx";

export interface SyncOptions<T = unknown> extends IAutorunOptions {
  fireImmediately?: boolean;
  equals?: IEqualsComparer<T>;
}
```

Reply:
Sure I can create a common one. If you look at their reasoning they wanted better typing support for explicit booleans in `fireImmediately`. Namely if it is `true` then they wanted the second parameter to be `T | undefined` but if it is `false` then to just have `T`.